ts/hooks/useHasPro.ts

```diff
@@ -1,17 +1,254 @@
+        // FIXME: set the sku dynamically if we dont use libsession
+        platform_link_cancel:
+          'https://play.google.com/store/account/subscriptions?package=network.loki.messenger&sku=SESSION_PRO_MONTHLY',
+        platform_link_refund: 'https://support.google.com/googleplay/workflow/9813244?',
+        session_support_link_refund: refundLinkSessionSupport,
+      };
+    case ProOriginatingPlatform.iOSAppStore:
+      return {
+        platform: 'Apple',
+        platform_account: 'Apple account',
+        platform_store: platformStoreApple,
+        platform_store_other: platformStoreGoogle,
+        device_type: 'iOS',
+        platform_link_manage: 'https://apps.apple.com/account/subscriptions',
+        platform_link_cancel: 'https://account.apple.com/account/manage/section/subscriptions',
+        platform_link_refund: 'https://support.apple.com/118223',
+        session_support_link_refund: refundLinkSessionSupport,
+      };
+    case ProOriginatingPlatform.Nil:
+      return {
+        platform: '',
+        platform_account: '',
+        platform_store: platformStoreGoogle,
+        platform_store_other: platformStoreApple,
+        device_type: '',
+        platform_link_manage: '',
+        platform_link_refund: '',
+        platform_link_cancel: '',
+        session_support_link_refund: refundLinkSessionSupport,
+      };
+    default:
+      return assertUnreachable(platform, `Unknown pro originating platform: ${platform}`);
+  }
+}
+
+function useMockProAccessExpiry() {
+  const variant = useDataFeatureFlag('mockProAccessExpiry') ?? MockProAccessExpiryOptions.MONTH;
+
+  // NOTE: for testing the expiry time should be pinned to x time after "now"
+  const now = variant ? Date.now() : 0;
+  switch (variant) {
+    case MockProAccessExpiryOptions.SOON:
+      return now + 600 * 1000;
+    case MockProAccessExpiryOptions.TODAY:
+      return now + 12 * 60 * 60 * 1000;
+    case MockProAccessExpiryOptions.TOMORROW:
+      return now + 26 * 60 * 60 * 1000;
+    case MockProAccessExpiryOptions.WEEK:
+      return now + 8 * 24 * 60 * 60 * 1000;
+    case MockProAccessExpiryOptions.MONTH:
+      return now + 30 * 24 * 60 * 60 * 1000;
+    case MockProAccessExpiryOptions.THREE_MONTH:
+      return now + 90 * 24 * 60 * 60 * 1000;
+    case MockProAccessExpiryOptions.YEAR:
+      return now + 12 * 30 * 24 * 60 * 60 * 1000;
+    case MockProAccessExpiryOptions.P24DT1M:
+      return now + 24 * 24 * 60 * 60 * 1000 + 60 * 60 * 1000;
+    case MockProAccessExpiryOptions.PT24H1M:
+      return now + 24 * 60 * 60 * 1000 + 60 * 60 * 1000;
+    case MockProAccessExpiryOptions.PT23H59M:
+      return now + 23 * 60 * 60 * 1000 + 59 * 60 * 1000;
+    case MockProAccessExpiryOptions.PT33M:
+      return now + 33 * 60 * 1000;
+    case MockProAccessExpiryOptions.PT1M:
+      return now + 1 * 60 * 1000;
+    case MockProAccessExpiryOptions.PT10S:
+      return now + 10 * 1000;
```

Comment:
I guess this is part of the PRD,

but if current local time is 10pm, and my pro expires in 10h, this will say "TODAY".
Is it expected?